### PR TITLE
Waggle-Sanity-Check v2, more tests, service waits

### PR DESCRIPTION
**Pull request by ozorob2**

### CODE REVIEW

File: ROOTFS/etc/waggle/sanity/fatal/agent-present.test [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+echo "Agent Present Test: Begin"
+
+expectedAgent=$(cat /etc/waggle/node_manifest.json | jq '[.nxagent | select(.present == true)] | length')
+
+if [[ $expectedAgent == 0 ]]; then
+    echo "Agent Present Test: No agent expected for this device PASS"
+    exit 0
+fi
+
+nmapOut=$(nmap -sP 10.31.81.1/24)
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/fatal/agent-present.test:12]: The agent is set to come up as 10.31.81.3 via our dnsmasq rules.  maybe we want to test for specifically that?

File: ROOTFS/etc/waggle/sanity/fatal/cameras-present.test [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/fatal/cameras-present.test:1]: @seanshahkarami and @gemblerz please coordinate on this.  this test does a test on the count of cameras detected on the network.  and the change that Yongho is working (https://github.com/waggle-sensor/waggle-sanity-check/pull/2) does something similar.

I think its is appropriate to have multiple camera tests.  1 (like this one) that just detects if the cameras are present on the network and a 2nd (Yongho's) that actually tests if the cameras work.

Just please coordinate.

File: ROOTFS/etc/waggle/sanity/fatal/network_switch.test [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#
+# NOTE(sean) is anyone changes the password to be different from the username,
+# this will fail and require more intellegence to estimate the current state.
+repeat 6 echo "logout" > "$dev"
+repeat 6 echo "ubnt" > "$dev"
+repeat 3 echo "show hardware" > "$dev"
+
+if ! timeout 10 grep -i -m 1 "software version" "$dev"; then
+    echo "Network Switch Test: could not detect version FAIL"
+    exit 1
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/fatal/network_switch.test:35]: please use different error code here `exit 2`

File: ROOTFS/etc/waggle/sanity/fatal/network_switch.test [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+if ! dev=$(ls /dev/serial/by-id/usb-FTDI* | head -n 1); then
+    echo "Network Switch Test: no serial console found FAIL"
+    exit 1
+fi
+
+# setup serial so we can interact with standard io tools
+stty -F $dev 115200 brkint -icrnl -imaxbel -opost -onlcr -isig -icanon -echo min 100 time 2
+
+# clear any buffered data
+timeout 3 cat "$dev" > /dev/null
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/fatal/network_switch.test:21]: just checking that that if you experience a timeout here (`timeout` returns an error) you want to keep running.  This is just a flush right?

File: ROOTFS/etc/waggle/sanity/config.ini [hunk 5]
```diff
@@ -3,7 +3,8 @@ interactive_tests = /etc/waggle/sanity/interactive/
 fatal_tests = /etc/waggle/sanity/fatal/
 warning_tests = /etc/waggle/sanity/warning/
 check_mins = 60
-warning_fail_led = [ "/sys/class/leds/wm-green3/", "/sys/class/leds/wm-red3/" ]
+warning_fail_led = [ "/sys/class/leds/wm-blue3/" ]
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/config.ini:6]: update to use a true orange color for "warning"

**Review by jswantek — CHANGES_REQUESTED** [review]

File: ROOTFS/etc/waggle/sanity/fatal/rpi-present.test [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+echo "RPi Present Test: Begin"
+
+expectedRPi=$(cat /etc/waggle/node_manifest.json | jq '[.shield | select(.present == true)] | length')
+
+if [[ $expectedRPi == 0 ]]; then
+    echo "RPi Present Test: No shield expected for this device PASS"
+    exit 0
+fi
+
+nmapOut=$(nmap -sP 10.31.81.1/24)
```
Comment (jswantek) [inline ROOTFS/etc/waggle/sanity/fatal/rpi-present.test:12]: The rpi is set to come up as 10.31.81.4 via our dnsmasq rules.  maybe we additionally want to test for specifically that?

File: ROOTFS/etc/waggle/sanity/fatal/network_switch.test (hunk 3 above)
Comment (seanshahkarami) [reply ROOTFS/etc/waggle/sanity/fatal/network_switch.test]: done

File: ROOTFS/etc/waggle/sanity/fatal/network_switch.test (hunk 4 above)
Comment (seanshahkarami) [reply ROOTFS/etc/waggle/sanity/fatal/network_switch.test:21]: That's right - this is intended to flush any data. (Without the timeout, cat reads forever.)

File: ROOTFS/etc/waggle/sanity/fatal/cameras-present.test (hunk 2 above)
Comment (seanshahkarami) [reply ROOTFS/etc/waggle/sanity/fatal/cameras-present.test:1]: Sure. Yeah, looks like Omar already has something working for detecting if all are present.

Just for the fact you have limited space to encode your results, I'd agree that keeping those checks split seems reasonable.

**Pull request merged**